Authentication integration

elezersolomon opened the pull request:
- [x]  cofigure code base structure.
- [x]  add api client
- [x] add login service
- [x] create tests for api client and login service
- [x] handled token storage
- [x] updated readme

elezersolomon committed:
feat(login and signup): login page enhacement and full sign up page

validation and ui improvement for login page, funn sign up page with code entry popup and redirect
to login page

elezersolomon committed:
added .git ignore to .gitignore

elezersolomon committed:
feat(adds tab template and enhances login and signup functionalities): ui enhancement & feature

removed headers for signup and login pages as well as new ones and added pages with tabs after login

elezersolomon committed:
docs(file inclusion): included git ignore

git ignore was included in git ignore it self and was untracked so i removed git ignore from git
ignore it self.

elezersolomon committed:
Update .gitignore

elezersolomon committed:
refactor(remove feature): removed signup feature

removed sign up feature as per comment on the current pull request.

#5, #6, #7

elezersolomon committed:
refactor(code base structure change): removed "screens" folder from repo

moved content from "screens" folder to "app" folder and removed "screens" folder

elezersolomon committed:
refactor(authentication services setup): setup login service, api client,  token storage and tests

changed login page credentials from username and password, to email and password, created login
service, created api client with get, post, put and delete functions, created tests with jest for
api client and login service, prepared .env store inviroment specific variables and included in
readme file.

#3

elezersolomon committed:
test(test file update): updated loginService testing method

updated loginService to use mocked axios instead of mocked api client to beter simulate backend api.

#3

elezersolomon committed:
fix(type declaration and cleanup): created .d.ts file for @env  error when emported and removed logs

declared types.d.ts file for @env error when emport and removed console log in login service file.

#3

elezersolomon committed:
docs(updated readme): added previously removed change

re included learn more and joing them comunity parts in readme file.

tsegaSolomon commented on app/login.tsx:
```diff
@@ -1,27 +1,84 @@
-    console.log('Login button pressed', username, password);
-  }
+    if (validateInputs()) {
+      loginUser(email, password);
```
Do we need error handling here?

elezersolomon replied:
error handling done for 401 and missing token senarios

tsegaSolomon commented on services/authService.ts:
This looks like a wrapper for AsyncStorage, may be rename it to tokenService, authService should do a lot more than that.

elezersolomon replied:
done

tsegaSolomon commented on services/loginService.ts:
may be rename this to authService and have all auth functions like login and logout in it.

elezersolomon replied:
done

tsegaSolomon reviewed (commented):
Please find half an hour to go through the code.
I think the authentication flow needs an update.

- Return user to login page if token is not found.
- Return user to login page if response is 401.

elezersolomon committed:
feat(code structure, new modules, code change): add error handling for 401 responce &  missing token

as per the comments in the previous pull request i have changed the file names and contents as
sugested, created  error handling for login page for un received token and invalid credentials
responces and updated coresponding tests, created home page, created function to get home page data
from api and created test for the function. added login that returns error and logges usre out on
missing token on HttpContext module.

BREAKING CHANGE: login will not work unless fake token is set in code for the time being.

#4

elezersolomon commented:
> Please find half an hour to go through the code. I think the authentication flow needs an update.
> 
> * Return user to login page if token is not found.
> * Return user to login page if response is 401.

made it so that any request other than login page will be returned to login page if there is no token, and unable to login to home page if 401 is returned on login.
